src/metabase/driver/druid/query_processor.clj

```diff
@@ -84,7 +84,7 @@
      ::total              (merge defaults {:queryType :timeseries})
      ::grouped-timeseries (merge defaults {:queryType :timeseries})
      ::topN               (merge defaults {:queryType :topN
-                                           :threshold i/absolute-max-results})
+                                           :threshold 1000})
```

Comment:
👎  to replacing a constant definition with a magic number in code. We have an interface (the `i` in the snippet above) with constants for these kinds of limits.

More than happy to limit topN results, as I think you're identifying a clear problem that makes a lot of sense to address in this way.